sources/gateway.move

```diff
@@ -187,6 +206,33 @@ entry fun issue_withdraw_and_whitelist_cap(
     transfer::transfer(whitelist_cap, tx_context::sender(ctx));
 }
 
+// set_message_context sets the message context for authenticated call
+// it is called automatically by the tss immediately before 'on_call'
+entry fun set_message_context(message_context: &mut MessageContext, sender: String, target: address) {
+    assert!(evm::is_valid_evm_address(sender), EInvalidSenderAddress);
+    assert!(target != @0x0, EInvalidReceiverAddress);
+
+    message_context.sender = sender;
+    message_context.target = target;
+}
+
+// reset_message_context resets the message context for authenticated call
+// it is called automatically by the tss immediately after 'on_call'
+entry fun reset_message_context(message_context: &mut MessageContext) {
```

Comment:
Only the TSS should be able to call these methods.

&WithdrawCap should be passed as an argument

Reply:
The `MessageContext` object self is an owned object (like the WithdrawCap). After issuing, the TSS will take the owner of the message context object, so I think the `set_message_context ` and `reset_message_context` can only be called by TSS self because the object self is a cap.

Also, when TSS passes it to `on_call` function, the object is passed as immutable reference [here](https://github.com/zeta-chain/node/blob/develop/zetaclient/chains/sui/signer/withdraw_and_call.go#L396). So nobody can actually modify the object except TSS.

Do we really need the `WithdrawCap` to protect MessageContext (self is a cap)? @lumtis